Source/Scene/getPolygonClippingFunction.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Obtain the GLSL fragment shader functions for performing the
+ * clipping polygon operations.
+ *
+ * @returns {String} A string containing GLSL functions for
+ * retrieving the clipping polygon data textures.
+ * @private
+ */
+
+function getClippingPolygonFunction(union) {
+  return (
+    "  uniform vec3 u_clippingPolygonBoundingBox[4];\n" +
+    "  uniform vec2 u_clippingPolygonCellDimensions;\n" +
+    "  uniform sampler2D u_clippingPolygonAccelerationGrid;\n" +
+    "  uniform sampler2D u_clippingPolygonMeshPositions;\n" +
+    "  uniform sampler2D u_clippingPolygonOverlappingTriangleIndices;\n" +
+    "  uniform vec2 u_clippingPolygonAccelerationGridPixelDimensions;\n" +
+    "  uniform vec2 u_clippingPolygonOverlappingTrianglePixelIndicesDimensions;\n" +
+    "  uniform vec2 u_clippingPolygonMeshPositionPixelDimensions;\n" +
+    "  uniform mat4 u_clippingPolygonEyeToWorldToENU;\n" +
+    "  uniform float u_clippingPolygonMinimumZ;\n" +
```

Comment:
Try to pack uniforms wherever possible since `gl.uniform` related functions happen to be a huge expense if you look at the profiler for any CeisumJS app. Looks like zoom of the `vec2`'s can be packed at least.